spec/compiler/crystal/tools/doc/type_spec.cr

```diff
@@ -97,25 +97,45 @@ describe Doc::Type do
       foo = generator.type(program.types["Foo"])
       foo.node_to_html("Foo".path(global: true)).should eq(%(<a href="Foo.html">Foo</a>))
     end
+  end
 
-    it "ASTNode has no superclass" do
-      program = semantic(<<-CODE).program
-        module Crystal
-          module Macros
-            class ASTNode
-            end
-            class Arg < ASTNode
-            end
+  it "ASTNode has no superclass" do
+    program = semantic(<<-CODE).program
+      module Crystal
+        module Macros
+          class ASTNode
+          end
+          class Arg < ASTNode
           end
         end
-        CODE
+      end
+      CODE
 
-      generator = Doc::Generator.new program, [""]
-      macros_module = program.types["Crystal"].types["Macros"]
-      astnode = generator.type(macros_module.types["ASTNode"])
-      astnode.superclass.should eq(nil)
-      # Sanity check: subclasses of ASTNode has the right superclass
-      generator.type(macros_module.types["Arg"]).superclass.should eq(astnode)
-    end
+    generator = Doc::Generator.new program, [""]
+    macros_module = program.types["Crystal"].types["Macros"]
+    astnode = generator.type(macros_module.types["ASTNode"])
+    astnode.superclass.should eq(nil)
+    # Sanity check: subclasses of ASTNode has the right superclass
+    generator.type(macros_module.types["Arg"]).superclass.should eq(astnode)
+  end
+
+  it "ASTNode has no ancestors" do
+    program = semantic(<<-CODE).program
+      module Crystal
+        module Macros
+          class ASTNode
+          end
+          class Arg < ASTNode
+          end
+        end
+      end
+      CODE
+
+    generator = Doc::Generator.new program, [""]
+    macros_module = program.types["Crystal"].types["Macros"]
+    astnode = generator.type(macros_module.types["ASTNode"])
+    astnode.ancestors.empty?.should be_true
```

Comment:
Maybe you could just copy this line to the previous example? The rest seems pretty much identical and I don't think we need a separate spec for this. Ancestors and superclass can be tested as one.

Reply:
Two specs looking similar is not justification for merging them since the failures then cannot be distinguished (especially since `ancestors` does not actually depend on `superclass`). In fact I'd go as far as splitting the negative and positive (sanity) tests into two.

Reply:
I don't this necessary. It just blows up. If any expectation fails, the entire example fails. In that case, you may need to investigate further which ones are affected and which are not. But this is super rare.
I see no point in duplicating lots of boilerplate for having highly isolated examples.

Reply:
> But this is super rare.

We cannot assume a priori that this is the case just because all specs would always ultimately pass. Besides there really isn't a lot of boilerplate here compared to many other compiler specs.